File: Fundamentals-of-Programing141-main/in-class/09_other_types/phone_number.py
"""
In-Class Exercise: Write a phone number parser that accepts a 10-digit
                   number of the form DDDDDDDDDD, and prints out:
                       a. number of digits
                       b. formatted (DDD) DDD-DDDD version
                       c. area code only
                       e. local number only

 File Name: phone_number.py
 Course:    CPTR 141
"""

# collect user input
phone = input("Enter a phone number: ")

# replace the 0 and x's below with appropriate string opererations
print("You enered {} digits, the first was \"{}\"".format(
  len(phone),
  phone[0]
))
print("Formatted: ({}) {}-{}".format(
  phone[0:3],
  phone[3:6],
  phone[6:]
))
print("Area Code: {}".format(
  phone[0:3]
))
print("Local Form: {}-{}".format(
  phone[3:6],
  phone[6:]
))

File: Fundamentals-of-Programing142-main/in-class/11_cpp_non-numeric_types/phone_number.cpp
/*************************************************************************
 *
 * In-Class Exercise: Write a phone number parser that accepts a 10-digit
 *                    number of the form DDDDDDDDDD, and prints out:
 *                       a. number of digits
 *                       b. formatted (DDD) DDD-DDDD version
 *                       c. area code only
 *                       e. local number only
 *
 * File Name: phoneNumber.cpp
 * Course:    CPTR 142
 *
 */

#include <iostream>
#include <string>
//using namespace std;
int main() {

  // We use a single string variable in this exercise
  std::string phone;

  // Prompt the user for input
  std::cout << "Enter a phone number: ";
  std::cin >> phone;

  // Display information
  std::cout << "You entered " << phone.length() << "digits, the fist was \"" << phone.at(0)<< "\"." << std::endl;  
  // Format the phone number
  std::cout << "Formatted: (" << phone.substr(0, 3) << ") " << phone.substr(2,3) << "-" << phone.substr(6, 4);
  
  // Display the various parts

  return 0;
}
